ui/tests/acceptance/task-fs-test.js

```diff
@@ -37,4 +43,127 @@ module('Acceptance | task fs', function(hooks) {
       'Empty state explains the condition'
     );
   });
+
+  test('visiting /allocations/:allocation_id/:task_name/fs/:path', async function(assert) {
+    const paths = ['some-file.log', 'a/deep/path/to/a/file.log', '/', 'Unicode™®'];
+
+    const testPath = async filePath => {
+      await FS.visitPath({ id: allocation.id, name: task.name, path: filePath });
+      assert.equal(
+        currentURL(),
+        `/allocations/${allocation.id}/${task.name}/fs/${encodeURIComponent(filePath)}`,
+        'No redirect'
+      );
+      assert.equal(FS.breadcrumbsText, `${task.name} ${filePath.replace(/\//g, ' ')}`.trim());
+    };
+
+    await paths.reduce(async (prev, filePath) => {
+      await prev;
+      return testPath(filePath);
+    }, Promise.resolve());
+  });
+
+  test('navigating allocation filesystem', async function(assert) {
+    await FS.visitPath({ id: allocation.id, name: task.name, path: '/' });
+
+    assert.ok(FS.fileViewer.isHidden);
+
+    assert.equal(FS.directoryEntries.length, 4);
+
+    assert.equal(FS.breadcrumbsText, task.name);
+
+    assert.equal(FS.breadcrumbs.length, 1);
+    assert.ok(FS.breadcrumbs[0].isActive);
+    assert.equal(FS.breadcrumbs[0].text, 'task-name');
+
+    FS.directoryEntries[0].as(directory => {
+      assert.equal(directory.name, 'directory', 'directories should come first');
+      assert.ok(directory.isDirectory);
+      assert.equal(directory.size, '', 'directory sizes are hidden');
+      assert.equal(directory.lastModified, 'a year ago');
+    });
+
+    FS.directoryEntries[2].as(file => {
+      assert.equal(file.name, '🤩.txt');
+      assert.ok(file.isFile);
+      assert.equal(file.size, '1 KiB');
+      assert.equal(file.lastModified, '2 days ago');
+    });
+
+    await FS.directoryEntries[0].visit();
+
+    assert.equal(FS.directoryEntries.length, 1);
+
+    assert.equal(FS.breadcrumbs.length, 2);
+    assert.equal(FS.breadcrumbsText, 'task-name directory');
+
+    assert.notOk(FS.breadcrumbs[0].isActive);
+
+    assert.equal(FS.breadcrumbs[1].text, 'directory');
+    assert.ok(FS.breadcrumbs[1].isActive);
+
+    await FS.directoryEntries[0].visit();
+
+    assert.equal(FS.directoryEntries.length, 1);
+    assert.notOk(FS.directoryEntries[0].path.includes('//'));
```

Comment:
What's the purpose of this assertion?

Furthermore, can you provide the third argument to the assertion so the explanation is captured?

Reply:
👍🏻 done, thanks, and for the other places this happens. This fiddly business with paths makes me wish for something like Rails’s `Pathname` 😐